src/kv/LevelDBStore.cc

```diff
@@ -11,6 +11,35 @@ using std::string;
 #include "common/debug.h"
 #include "common/perf_counters.h"
 
+#define dout_subsys ceph_subsys_leveldb
+#undef dout_prefix
+#define dout_prefix *_dout << "leveldb: "
+
+class CephLevelDBLogger : public leveldb::Logger {
+  CephContext *cct;
+public:
+  CephLevelDBLogger(CephContext *c) : cct(c) {
+    cct->get();
+  }
+  ~CephLevelDBLogger() {
+    cct->put();
+  }
+
+  // Write an entry to the log file with the specified format.
+  void Logv(const char* format, va_list ap) {
+    dout(1);
+    char buf[1048576];
```

Comment:
Hmm, I would like to think 1M is a dangerous value to stack size? AFAR, some OS may only have several hundreds stack size.

Maybe we can declare a 1k buf, then try to iterate the whole line. If failed, turn to heap allocation? Or just truncate the log line.